### PR TITLE
Adding tests for http proxy sending 504 status code

### PR DESCRIPTION
**Pull request by jitsni**

### CODE REVIEW

**Comment by jitsni** [comment]
@MariusLeahu good work. Creating a PR based on your work.

Do NOT merge yet, it needs k3po release.

**Comment by jitsni** [comment]
One test perhaps missing is : when server takes too much time, then proxy should send 504. May be it can be addressed when DOS is addressed.

File: pom.xml [hunk 1]
```diff
@@ -28,7 +28,7 @@
         <maven.compiler.testSource>1.8</maven.compiler.testSource>
         <maven.compiler.testTarget>1.8</maven.compiler.testTarget>
         <hazelcast.version>1.9.4.8</hazelcast.version>
-        <k3po.version>2.1.0</k3po.version>
+        <k3po.version>develop-SNAPSHOT</k3po.version>
```
Comment (jitsni) [inline pom.xml]: TODO need k3po release

File: service/http.proxy/pom.xml [hunk 2]
```diff
@@ -67,9 +67,10 @@
             <classifier>keystore</classifier>
             <scope>test</scope>
         </dependency>
-        <dependency>
+         <dependency>
             <groupId>org.kaazing</groupId>
             <artifactId>specification.http</artifactId>
+            <version>develop-SNAPSHOT</version>
```
Comment (jitsni) [inline service/http.proxy/pom.xml]: TODO delete the version

**Comment by dpwspoon** [comment]
These needs a new release of K3po

**Comment by jitsni** [comment]
This is ready for merge.

**Comment by dpwspoon** [comment]
@jitsni Sorry, this one slipped my notice.  Can you do a catch up merge and then ping me on slack or here and I'll merge it

**Comment by jitsni** [comment]
@dpwspoon Done the catchup merge.

**Pull request merged**